enhancements/insights/rapid-recommendations.md

```diff
@@ -0,0 +1,549 @@
+and gather data specified in the configuration. 
+The Observability Intelligence team (former CCX) will maintain the remote configuration in a repository 
+and deploy it to console.redhat.com whenever it changes.
+
+### Workflow Description
+
+This proposal introduces a new workflow for requesting new data to be collected by the Insights Operator.
+
+A **data requester** is a person who analyzes or develops tools to analyze Insights Operator archives
+uploaded to Red Hat.
+
+1. A data requester identifies the need for new data in the Insights Operator archive.
+2. The data requester describes the data using a format required by the remote configuration service.
+3. The data requester creates a pull request to the repository where the remote configuration is maintained.
+4. The pull request is reviewed (and either approved or denied) 
+   by a member of the Observability Intelligence (former CCX) team.
+5. When the pull request is merged, the new remote configuration is deployed 
+   to the remote configuration service on console.redhat.com.
+6. Insights Operators in connected clusters download the new remote configuration 
+   and start collecting the new data.
+7. The data requester makes use of the new data in newly incoming Insights Operator archives.
+
+### API Extensions
+
+This proposal does not add any API extension.
+
+### Topology Considerations
+
+#### Hypershift / Hosted Control Planes
+
+This proposal does not require any specific details to account for the HyperShift. 
+For now, we can simply extend the remote configuration file to 
+request gathering of specific HyperShift resources.
+
+#### Standalone Clusters
+
+This proposal is relevant to standalone (and especially connected) clusters. 
+Standalone cluster retrieves the remote configuration (based on its OCP version) and collects data accordingly. 
+
+#### Single-node Deployments or MicroShift
+
+This proposal may lead to increased resource consumption by the Insights Operator 
+(described in the [Excessive use of cluster resources](#excessive-use-of-cluster-resources) section), 
+but the operator can be disabled in these environments or in case of problems. 
+
+### Implementation Details/Notes/Constraints 
+
+#### Remote Configuration Service
+
+The Insights Operator configuration will replace the current [conditional gathering endpoint](../insights/conditional-data-gathering.md)
+
+```bash
+https://console.redhat.com/api/gathering/gathering_rules.json
+```
+
+with a new one:
+
+```bash
+https://console.redhat.com/api/gathering/v2/%s/gathering_rules.json
+```
+
+The `%s` parameter is the OCP version of the Insights Operator as set in the `.status.versions` map 
+of the Insights `clusteoperator.config.openshift.io` resource (consider partially upgraded clusters).
+The remote configuration service will be responsible for serving configuration that can be 
+successfully processed by an Insights Operator of the given version.
```

Comment:
I'd recommend the operator use _its_ version, which [you request as `RELEASE_VERSION`][1].  During an update:

1. CVO working through the update...
2. CVO gets to your `06-deployment.yaml` manifiest, and bumps the Deployment to request your new operator image, setting a new `RELEASE_VERSION`, etc.
3. Deployment controller sees your [`type: Recreate`][2], and scales down the outgoing Pod.
4. Deployment controller scales up an incoming Pod.
5. Incoming Insights operator container comes up, notices that ClusterOperator's `status.versions` is older than its own `RELEASE_VERSION`, and starts updating your component.  Maybe there's not all that much going on in the Insights component outside of the operator Deployment?
6. Insights operator decides the whole component has updates, and bumps `status.versions[name=operator]` in your ClusterOperator.

During step 5, it's the incoming Insights operator that's running, and I think you want to be requesting incoming-version rules then, instead of having the incoming operator initially requesting outgoing-version rules.  But if you always get to step 6 really fast, and will never poll for rules before that point, there's no distinction.

> And if an upgrade is started after version is read but before collection has been finished?

With my suggestion here, each Insights operator container will only ever request its version of rules, so there's no room for version-skew between the rules and the operator code.  There is room for skew between the rules/operator and the rest of the cluster though.  E.g. in an A->B update, by the time the vB Insights operator rolls in level 50, [the Kube API server bits will have all reached vB][3], but network, DNS, and machine-config components will not have updated yet.  So if the gathering rule asks for some new-in-vB network pod (say), that pod might not exist for a few minutes early in the new Insights operator's life.

[1]: https://github.com/openshift/insights-operator/blob/d00f74fdef22f00c3ec5ed8a8e2942a7afb5a6e9/manifests/06-deployment.yaml#L98-L99
[2]: https://github.com/openshift/insights-operator/blob/d00f74fdef22f00c3ec5ed8a8e2942a7afb5a6e9/manifests/06-deployment.yaml#L14
[3]: https://docs.openshift.com/container-platform/4.15/updating/understanding_updates/understanding-openshift-update-duration.html#update-duration-example_openshift-update-duration